lib/routers/establishment/invite-user.js

```diff
@@ -28,10 +28,27 @@ const validateInvitation = (req, res, next) => {
   })(req, res, next);
 };
 
+const preventDuplicateInvite = (req, res, next) => {
+  return Promise.resolve()
+    .then(() => {
+      return req.models.Profile.query()
+        .scopeToEstablishment('establishments.id', req.establishment.id)
+        .where('profiles.email', req.body.data.email)
+        .first();
+    })
+    .then(profile => {
+      if (profile) {
+        next(new Error(`This user is already associated with ${req.establishment.name}`));
+      }
+    })
+    .then(() => next());
```

Comment:
You probably want to throw the error rather than calling `next` with it above, as that will allow promise chain execution to continue.

Throw the error and then add a `.catch` at the bottom of the chain (which is also needed to handle any other possible errors).

I'd also probably set a status on the error being thrown so it doesn't report 500 - 4xx of some kind, or maybe even a 304 if we're feeling particularly wild, although that might be harder to catch on the client.